core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/collparam/semantics/CollectionSemantics.java

```diff
@@ -18,119 +18,51 @@
  */
 package org.apache.isis.core.metamodel.facets.collparam.semantics;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.Queue;
-import java.util.Set;
-import java.util.SortedSet;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
 
-import org.apache.isis.commons.collections.Can;
+import org.apache.isis.core.config.progmodel.ProgrammingModelConstants;
 
-public enum CollectionSemantics {
+public interface CollectionSemantics {
 
-    ARRAY(true),
+    // -- FACTORIES
 
-    COLLECTION_INTERFACE(true),
```

Comment:
reinstate enum names via toString for metamodel export; or replace by entirely by CollectionType from ProgrammingModelConstants